src/handler.rs

```diff
@@ -105,13 +118,28 @@ impl axum::response::IntoResponse for Error {
     }
 }
 
+type Result<T> = std::result::Result<T, Error>;
+
+fn get_server_from_state<'a>(
+    state: &'a OPRFState,
+    instance_name: &'a str,
+) -> Result<RwLockReadGuard<'a, OPRFInstance>> {
+    Ok(state
+        .instances
+        .get(instance_name)
+        .ok_or_else(|| Error::InstanceNotFound(instance_name.to_string()))?
+        .read()?)
+}
+
 /// Process PPOPRF evaluation requests
-pub async fn randomness(
-    State(state): State<OPRFState>,
-    Json(request): Json<RandomnessRequest>,
-) -> Result<Json<RandomnessResponse>, Error> {
+#[instrument(skip(state, request))]
+async fn randomness(
```

Comment:
If there are no `await` calls in here, this function could be synchronous, handling the full request in a single future. Might improve throughput, but I didn't check.